src/subliminal/providers/opensubtitlescom.py

```diff
@@ -595,30 +595,29 @@ def api_get(
 
         return r.json()  # type: ignore[no-any-return]
 
-    def _search(self, *, page: int = 1, **params: Any) -> list[dict[str, Any]]:
-        # Extended params with page
-        ext_params = {'page': page, **params}
-
-        # query the server
-        logger.info('Searching subtitles %r', ext_params)
-
-        # GET request and add page information
-        response = self.api_get('subtitles', ext_params)
-
-        if not response or not response['data']:
-            return []
-
-        ret = response['data']
+    def _search(self, **params: Any) -> Generator[dict[str, Any], None, None]:
+        page = 1
+
+        while True:
+            # Extended params with page
+            ext_params = {'page': page, **params}
+            logger.info('Searching subtitles %r', ext_params)
+            # GET request and add page information
+            response = self.api_get('subtitles', ext_params)
+            if not response or not response['data']:
+                break
+            else:
```

Comment:
This `else` can be removed and the `yield from` statement deindented

Reply:
Updated